File: 2020/julia/src/Day25.jl
module Day25
include("Aoc.jl")
using .Aoc

mod = 2020_12_27
(doorspublic, cardspublic) = parse.(Int64, Aoc.input_lines(25))

function part1()
    loop = 0
    while powermod(7, (loop += 1), mod) != cardspublic end
    powermod(doorspublic, loop, mod)
end
part2() = ""
end

File: 2020/typescript/src/days/day25.ts
import {Day} from "../aoc";

export class Day25 extends Day {
    day = () => 25;

    part1(): string {
        let [cardPubKey, doorPubKey, mod, key] = [...this.input.map(Number), 2020_12_27, 1];
        for (let loop = 1; loop !== cardPubKey; loop *= 7, loop %= mod) key = (key * doorPubKey) % mod;
        return key.toString();
    }

    part2 = (): string => '';
}